### PR TITLE
Repair Spectrogram after Data Refactor

### PR DESCRIPTION
**Pull request by retiutut**
- Remove "Save Spectrogram" button - not enough room, and GUI screenshot is better idea anyways
- Fix timestamps

### CODE REVIEW

**Comment by retiutut** [comment]
Let me see if I can update the code. The timestamp here is only used to display on the screen.

File: OpenBCI_GUI/W_Spectrogram.pde [hunk 1]
```diff
@@ -464,16 +425,14 @@ class W_Spectrogram extends Widget {
...
+        List<double[]> currentData = currentBoard.getData(1);
+        int timeStampChan = currentBoard.getTimestampChannel();
+        long timestampMS = (long)(currentData.get(0)[timeStampChan] * 1000.0);
+        if(timestampMS == 0) {
+            return "TimeNotFound";
         }
-        
-        return "TimeNotFound";
+        String output = new SimpleDateFormat("HH:mm:ss").format(timestampMS);
+        return output;
```
Comment (daniellasry) [inline OpenBCI_GUI/W_Spectrogram.pde:435]: Agreed with andrey. This function should not return a string, it should return a long. You can use your `DateTimeFormatter formatter` above to format it later.

File: OpenBCI_GUI/W_Spectrogram.pde [hunk 2]
```diff
@@ -429,27 +396,21 @@ class W_Spectrogram extends Widget {
...
+                time = LocalTime.parse(getCurrentTimeStamp());
             }
+        } catch (NullPointerException e) {
+            println("Spectrogram: Timestamp error...");
+            e.printStackTrace();
+            time = LocalTime.now();
+        } catch (NumberFormatException e) {
+            println("Spectrogram: Timestamp error...");
+            e.printStackTrace();
+            time = LocalTime.now();
```
Comment (daniellasry) [inline OpenBCI_GUI/W_Spectrogram.pde:413]: Instead of catching these exceptions, fix the actual bug they are hiding.

Comment (retiutut) [reply OpenBCI_GUI/W_Spectrogram.pde]: I think the data refactor + brainflow has mitigated these exceptions.

**Comment by retiutut** [comment]
Switching usage of `LocalTime` to `LocalDateTime` since the latter includes the following method, allowing use to input a `long` from currentBoard:

https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#ofInstant-java.time.Instant-java.time.ZoneId-

**Pull request merged**